Shared/AzureFunctions/New-JsonWebToken.ps1

```diff
@@ -0,0 +1,116 @@
+﻿# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+function New-JsonWebToken {
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSUseShouldProcessForStateChangingFunctions', '', Justification = 'Only non-destructive operations are performed in this function.')]
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory = $true)]
+        [string]$CertificateThumbprint,
+
+        [ValidateSet("CurrentUser", "LocalMachine")]
+        [Parameter(Mandatory = $false)]
+        [string]$CertificateStore = "CurrentUser",
+
+        [Parameter(Mandatory = $false)]
+        [string]$Issuer,
+
+        [Parameter(Mandatory = $false)]
+        [string]$Audience,
+
+        [Parameter(Mandatory = $false)]
+        [string]$Subject,
+
+        [Parameter(Mandatory = $false)]
+        [string]$TokenLifetimeInSeconds = 3600,
+
+        [ValidateSet("RS256", "RS384", "RS512")]
+        [Parameter(Mandatory = $false)]
+        [string]$SigningAlgorithm = "RS256"
+    )
+
+    <#
+        Shared function to create a signed Json Web Token (JWT) by using a certificate.
+        It is also possible to use a secret key to sign the token, but that is not supported in this function.
+        The function returns the token as a string if successful, otherwise it returns $null.
+        https://www.rfc-editor.org/rfc/rfc7519
+        https://learn.microsoft.com/azure/active-directory/develop/active-directory-certificate-credentials
+        https://learn.microsoft.com/azure/active-directory/develop/v2-oauth2-client-creds-grant-flow
```

Comment:
This currently does NOT return a `$null` value the way you have it set up.

A return in `process` will execute the `end` block still. Then returning ".." base off this line:

```
    end {
        Write-Verbose "Returning Json Web Token"
        return ($headerBase64 + "." + $payloadBase64 + "." + $signature)
    }
```